crates/ln-dlc-node/examples/fund.rs

```diff
@@ -0,0 +1,223 @@
+extern crate ln_dlc_node;
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use bitcoin::Amount;
+use ln_dlc_node::node::NodeInfo;
+use local_ip_address::local_ip;
+use reqwest::Response;
+use serde::Deserialize;
+use std::time::Duration;
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::filter::Directive;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+
+const RUST_LOG_ENV: &str = "RUST_LOG";
+const FAUCET: &str = "http://localhost:8080";
+const COORDINATOR: &str = "http://localhost:8000";
+
+#[tokio::main]
+async fn main() {
+    init_tracing(LevelFilter::DEBUG).expect("tracing to initialise");
+    fund_everything().await.expect("to be able to fund");
+}
+
+async fn fund_everything() -> Result<()> {
+    let coord_addr = get_coordinator_address().await?;
+    fund(&coord_addr, Amount::ONE_BTC).await?;
+    mine(20).await?;
+
+    let coordinator_balance = get_text(&format!("{COORDINATOR}/api/balance")).await?;
+    tracing::info!("coordinator BTC balance: {}", coordinator_balance);
+
+    let node: NodeInfo = reqwest::get(format!("{COORDINATOR}/api/node"))
+        .await?
+        .json()
+        .await?;
+    tracing::info!("lightning node: {}", node);
+
+    let lnd_addr: LndAddr = reqwest::get(&format!("{FAUCET}/lnd/v1/newaddress"))
+        .await?
+        .json()
+        .await?;
+
+    fund(
+        &lnd_addr.address,
+        Amount::ONE_BTC
+            .checked_mul(2)
+            .expect("small integers to multiply"),
+    )
+    .await?;
+    mine(20).await?;
+
+    let lnd_balance = get_text(&format!("{FAUCET}/lnd/v1/balance/blockchain")).await?;
+    tracing::info!("coordinator lightning balance: {}", lnd_balance);
+
+    open_channel(
+        &node,
+        Amount::ONE_BTC
+            .checked_div(10)
+            .expect("small integers to divide"),
+    )
+    .await?;
+
+    let lnd_channels = get_text(&format!("{FAUCET}/lnd/v1/channels")).await?;
+    tracing::info!("open LND channels: {}", lnd_channels);
+    Ok(())
+}
+
+#[derive(Deserialize)]
+struct LndAddr {
+    address: String,
+}
+
+// Includes some bespoke text processing that ensures we can deserialise the response properly
+async fn get_coordinator_address() -> Result<String> {
+    Ok(get_text(&format!("{COORDINATOR}/api/newaddress"))
+        .await?
+        .strip_prefix('"')
```

Comment:
I think the problem is that we return the address string as a json. We could fix that and either return a type holding the address or just the address as string.